### PR TITLE
Shut down the Ollama service if there are no Ollama models to evaluate, to avoid unnecessary processes being opened

### PR DESCRIPTION
**Pull request by ruiAzevedo19**
Part of #225

### CODE REVIEW

**Comment by Munsio** [comment]
Out of curiosity I checked this PR and I have to say instead of shutting down the ollama server we shouldn't even start it if it is not needed.

Also @ruiAzevedo19 it is Issue #225 not #255

**Comment by bauersimon** [comment]
Yeah I think we should maybe reorder that the providers are parsed from the CLI first... and then we don't even start the ones that are not needed :thinking:

**Comment by bauersimon** [comment]
Well if there is no model that starts with `ollama/` we will never have an ollama model.

**Comment by ruiAzevedo19** [comment]
> Well if there is no model that starts with `ollama/` we will never have an ollama model.

We only add the `ollama/` prefix after fetching the Ollama models

**Comment by Munsio** [comment]
> > Well if there is no model that starts with `ollama/` we will never have an ollama model.
> 
> We only add the `ollama/` prefix after fetching the Ollama models

And that needs to be swapped. Parse the CLI first and only parse models of providers which are needed. Then do the validation.

**Review by Munsio — APPROVED** [review]

File: cmd/eval-dev-quality/cmd/evaluate_test.go [hunk 1]
```diff
@@ -525,6 +525,31 @@ func TestEvaluateExecute(t *testing.T) {
...
+
+					ExpectedResultFiles: map[string]func(t *testing.T, filePath string, data string){
+						filepath.Join("result-directory", "categories.svg"): nil,
+						filepath.Join("result-directory", "evaluation.csv"): nil,
+						filepath.Join("result-directory", "evaluation.log"): nil,
+						filepath.Join("result-directory", "README.md"):      nil,
+						filepath.Join("result-directory", string(evaluatetask.IdentifierWriteTests), "symflower_symbolic-execution", "golang", "golang", "plain.log"): nil,
+					},
+					ExpectedOutputValidate: func(t *testing.T, output, resultPath string) {
+						assert.NotContains(t, output, "Starting services for provider \"ollama\"")
```
Comment (Munsio) [inline cmd/eval-dev-quality/cmd/evaluate_test.go:551]: Sry for the nitpick but please change this to: 
```
assert.NotContains(t, output, `Starting services for provider "ollama"`)
```

The other one too.

**Review by Munsio — CHANGES_REQUESTED** [review]

**Review by Munsio — APPROVED** [review]

**Pull request merged**